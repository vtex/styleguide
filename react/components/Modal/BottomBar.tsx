import React, { forwardRef } from 'react'
import classNames from 'classnames'

export interface Props {
  showBorder?: boolean
  responsiveFullScreen?: boolean
  children?: React.ReactNode
}

function BottomBar(
  { showBorder = true, responsiveFullScreen = false, children }: Props,
  forwardedRef: React.Ref<HTMLDivElement>
) {
  if (!children) return <></>
  return (
    <div
      className={classNames(
        'flex justify-content flex-row-reverse min-h-regular-ns min-h-small pv6-ns ph8-ns pv7 ph6 mt3',
        {
          'bt b--muted-4': showBorder,
          pb7: responsiveFullScreen,
        }
      )}
      style={responsiveFullScreen ? { marginTop: 'auto' } : {}}
      ref={forwardedRef}
    >
      <div>{children}</div>
    </div>
  )
}

export default forwardRef<HTMLDivElement, Props>(BottomBar)
