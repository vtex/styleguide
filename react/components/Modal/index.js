import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import ResponsiveModal from 'react-responsive-modal'
import get from 'lodash/get'

import zIndex from '../utils/zIndex'
import TopBar from './TopBar'
import BottomBar from './BottomBar'
import styles from './modal.css'

import './modal.global.css'

class Modal extends PureComponent {
  constructor(props) {
    super(props)
    this.contentContainerReference = React.createRef()
    this.state = { shadowBottom: false, shadowTop: false }
  }

  componentDidMount() {
    this.setShadowState(this.contentContainerReference.current)
  }

  componentDidUpdate(prevProps) {
    const scrollTop = get(this, 'contentContainerReference.current.scrollTop')
    if (
      prevProps.isOpen !== this.props.isOpen ||
      (scrollTop === 0 && this.state.shadowTop)
    )
      this.setShadowState(this.contentContainerReference.current)
  }

  handleScroll = event => {
    this.setShadowState(event.target)
  }

  setShadowState = element => {
    if (!element) return

    const { scrollTop, scrollHeight, clientHeight } = element

    this.setState({ shadowTop: scrollTop !== 0 })
    this.setState({ shadowBottom: scrollHeight - scrollTop !== clientHeight })
  }

  render = () => {
    const {
      isOpen,
      centered,
      onClose,
      closeOnEsc,
      closeOnOverlayClick,
      showCloseIcon,
      bottomBar,
      title,
      children,
      responsiveFullScreen,
      showTopBar,
      showBottomBarBorder,
      onCloseTransitionFinish,
      container,
    } = this.props
    const { shadowBottom, shadowTop } = this.state

    return (
      <ResponsiveModal
        open={isOpen}
        center={centered}
        container={container}
        onClose={onClose}
        onExited={onCloseTransitionFinish}
        closeOnEsc={closeOnEsc}
        closeOnOverlayClick={closeOnOverlayClick}
        showCloseIcon={showTopBar ? false : showCloseIcon}
        classNames={{
          overlay: `vtex-modal__overlay ${
            responsiveFullScreen ? 'pa5-ns pa0' : ''
          }`,
          modal: `vtex-modal__modal ${
            responsiveFullScreen ? 'br2-ns w-100 h-100 h-auto-ns' : 'br2 w-100'
          } ${styles.mh100} flex flex-column`,
          closeIcon: 'vtex-modal__close-icon',
        }}
        styles={{
          overlay: {
            backgroundColor: 'rgba(0, 0, 0, 0.8)',
            zIndex: zIndex.modal,
          },
          modal: {
            padding: 0,
          },
          closeIcon: {
            top: '8px',
            right: '8px',
            padding: '10px',
          },
        }}
        closeIconSize={18}>
        {showTopBar && (
          <TopBar
            title={title}
            onClose={onClose}
            showBottomShadow={shadowTop}
            responsiveFullScreen={responsiveFullScreen}
            showCloseIcon={showCloseIcon}
          />
        )}
        <div
          className={`${
            responsiveFullScreen ? 'ph7 ph8-ns' : 'ph6 ph8-ns'
          } overflow-auto flex-shrink-1 flex-grow-1 ${bottomBar ? '' : 'pb8'} ${
            title ? '' : 'pt5 pt6-ns'
          } ${styles.scrollBar}`}
          ref={this.contentContainerReference}
          onScroll={this.handleScroll}>
          {children}
        </div>
        {bottomBar ? (
          <BottomBar
            showTopShadow={shadowBottom}
            showBorder={showBottomBarBorder}
            responsiveFullScreen={responsiveFullScreen}>
            {bottomBar}
          </BottomBar>
        ) : (
          ''
        )}
      </ResponsiveModal>
    )
  }
}

Modal.defaultProps = {
  isOpen: false,
  closeOnEsc: true,
  closeOnOverlayClick: true,
  showCloseIcon: true,
  showTopBar: true,
  showBottomBarBorder: true,
}

Modal.propTypes = {
  /** Content of the modal */
  children: PropTypes.node.isRequired,
  /** Center the modal (for small content) */
  centered: PropTypes.bool,
  /** Container in which the modal is rendered */
  container: PropTypes.object,
  /** Show or hide the modal */
  isOpen: PropTypes.bool.isRequired,

  onClose: PropTypes.func.isRequired,

  /** Show BottomBar border **/
  showBottomBarBorder: PropTypes.bool,
  /** Close the modal on ESC key press (default true) */
  closeOnEsc: PropTypes.bool,
  /** Close the modal on overlay click (default true) */
  closeOnOverlayClick: PropTypes.bool,
  /** Show the close icon on upper right corner (default true) */
  showCloseIcon: PropTypes.bool,
  /** Node to be displayed as the bottom bar of the modal. */
  bottomBar: PropTypes.node,
  /** Modal title to be displayed in top of the modal. */
  title: PropTypes.node,
  /** If true, the modal will expand to fullscreen in small view ports (e.g. mobile) */
  responsiveFullScreen: PropTypes.bool,
  /** If true, show top bar with title */
  showTopBar: PropTypes.bool,
  /** Event fired when the closing transition is finished */
  onCloseTransitionFinish: PropTypes.func,
}

export default Modal
