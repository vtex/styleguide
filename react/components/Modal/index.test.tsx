import React, { useRef } from 'react'
import { render, fireEvent } from '@testing-library/react'

import Modal from '.'
import Input from '../Input'

describe('Modal', () => {
  jest.spyOn(window, 'scroll').mockImplementation()

  it('should have a default export', () => {
    expect(typeof Modal).toBe('object')
  })

  describe('onClose', () => {
    it('should be called on click close button', () => {
      const onClose = jest.fn()

      const { getByRole } = render(
        <Modal isOpen onClose={onClose}>
          Foo
        </Modal>
      )

      const closeButton = getByRole('button')

      fireEvent.click(closeButton)

      expect(onClose).toHaveBeenCalled()
    })

    it('should be called on esc by default', () => {
      const onClose = jest.fn()

      const { getByRole } = render(
        <Modal isOpen onClose={onClose}>
          Foo
        </Modal>
      )

      fireEvent.keyDown(getByRole('dialog'), { key: 'Escape', keyCode: 27 })

      expect(onClose).toHaveBeenCalled()
    })

    it('should not be called when closeOnEsc prop is false', () => {
      const onClose = jest.fn()

      const { getByRole } = render(
        <Modal isOpen onClose={onClose} closeOnEsc={false}>
          Foo
        </Modal>
      )

      fireEvent.keyDown(getByRole('dialog'), { key: 'Escape', keyCode: 27 })

      expect(onClose).not.toHaveBeenCalled()
    })

    it('should be called when click on the overlay by default', () => {
      const onClose = jest.fn()

      const { container } = render(
        <Modal isOpen onClose={onClose}>
          Foo
        </Modal>
      )
      fireEvent.click(
        document.querySelector('[data-testid="modal__overlay"]') ?? container
      )
      expect(onClose).toHaveBeenCalled()
    })

    it('should not be called when closeOnOverlayClick prop is false', () => {
      const onClose = jest.fn()

      const { container } = render(
        <Modal isOpen onClose={onClose} closeOnOverlayClick={false}>
          Foo
        </Modal>
      )

      fireEvent.click(
        document.querySelector('[data-testid="modal__overlay"]') ?? container
      )
      expect(onClose).not.toHaveBeenCalled()
    })
  })

  describe('title', () => {
    it('should render title prop', () => {
      const containerModal = document.createElement('div')
      const onClose = jest.fn()
      const title = 'modal title'

      const { getByText, asFragment } = render(
        <Modal
          isOpen
          onClose={onClose}
          title={title}
          container={containerModal}
        >
          Foo
        </Modal>,
        { container: containerModal }
      )
      expect(getByText(title)).not.toBeNull()
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('isOpen', () => {
    it('Should show modal when isOpen is true and hide when is false', () => {
      const onClose = jest.fn()
      const containerModal = document.createElement('div')

      const { rerender, container } = render(
        <Modal isOpen onClose={onClose} container={containerModal}>
          Foo
        </Modal>,
        { container: containerModal }
      )
      expect(container).toMatchSnapshot()

      rerender(
        <Modal isOpen={false} onClose={onClose} container={containerModal}>
          Foo
        </Modal>
      )
      expect(container).toMatchSnapshot()
    })
  })

  describe('bottomBar', () => {
    it('should render bottomBar', () => {
      const onClose = jest.fn()
      const containerModal = document.createElement('div')

      const { getByText, container } = render(
        <Modal
          isOpen
          onClose={onClose}
          bottomBar={<div>Bar</div>}
          container={containerModal}
        >
          Foo
        </Modal>,
        { container: containerModal }
      )

      expect(getByText('Bar')).not.toBeNull()
      expect(container).toMatchSnapshot()
    })
  })

  describe('container', () => {
    it('should render modal inside it', () => {
      const onClose = jest.fn()
      const containerModal = document.createElement('div')
      const { container } = render(
        <Modal isOpen onClose={onClose} container={containerModal}>
          Foo
        </Modal>,
        { container: document.body.appendChild(containerModal) }
      )
      expect(
        container.querySelector('[data-testid="modal__modal"]')
      ).not.toBeNull()
      expect(container).toMatchSnapshot()
    })
  })

  describe('CSS API', () => {
    it('default', () => {
      const containerModal = document.createElement('div')
      const onClose = jest.fn()

      const { container } = render(
        <Modal isOpen onClose={onClose} container={containerModal}>
          Foo
        </Modal>,
        { container: containerModal }
      )

      expect(container).toMatchSnapshot()
    })

    it('responsiveFullScreen true', () => {
      const containerModal = document.createElement('div')
      const onClose = jest.fn()

      const { container } = render(
        <Modal
          isOpen
          onClose={onClose}
          responsiveFullScreen
          container={containerModal}
        >
          Foo
        </Modal>,
        { container: containerModal }
      )

      expect(container).toMatchSnapshot()
    })

    it('showTopBar false', () => {
      const containerModal = document.createElement('div')
      const onClose = jest.fn()

      const { container } = render(
        <Modal
          isOpen
          onClose={onClose}
          showTopBar={false}
          container={containerModal}
        >
          Foo
        </Modal>,
        { container: containerModal }
      )

      expect(container).toMatchSnapshot()
    })

    it('showBottomBarBorder false', () => {
      const containerModal = document.createElement('div')
      const onClose = jest.fn()

      const { container } = render(
        <Modal
          isOpen
          onClose={onClose}
          showBottomBarBorder={false}
          container={containerModal}
        >
          Foo
        </Modal>,
        { container: containerModal }
      )

      expect(container).toMatchSnapshot()
    })

    it('centered', () => {
      const containerModal = document.createElement('div')
      const onClose = jest.fn()

      const { container } = render(
        <Modal isOpen onClose={onClose} centered container={containerModal}>
          Foo
        </Modal>,
        { container: containerModal }
      )

      expect(container).toMatchSnapshot()
    })
  })

  describe('initialFocusRef', () => {
    it('to focus correctly', () => {
      const containerModal = document.createElement('div')
      const onClose = jest.fn()

      const Component = () => {
        const initialFocusRef = useRef<HTMLDivElement>(null)

        return (
          <Modal
            isOpen
            onClose={onClose}
            container={containerModal}
            initialFocusRef={initialFocusRef}
          >
            <Input
              placeholder="Type your name..."
              size="small"
              ref={initialFocusRef}
            />
          </Modal>
        )
      }
      render(<Component />, { container: containerModal })

      const currentFocused = document.activeElement

      expect(currentFocused).toBe(document.querySelector('input'))
    })
  })
})
