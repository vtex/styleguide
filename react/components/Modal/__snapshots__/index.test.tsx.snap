// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal CSS API centered 1`] = `
<div>
  <div
    class="flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill items-center"
    data-testid="modal__overlay"
    tabindex="-1"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      class="undefined"
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="vtex-modal__title"
        aria-modal="true"
        class="flex flex-column relative bg-white shadow-5 center mv9 vw-50-ns vw-70"
        data-testid="modal__modal"
        role="dialog"
      >
        <div
          class="mb5"
        >
          <div
            class="min-h-small min-h-large-ns pl6 pl8-ns"
          >
            <div
              class="fr pointer pr6-ns pt6-ns pr5 pt5"
              role="button"
              tabindex="0"
            >
              <svg
                class="vtex__icon-close  "
                fill="none"
                height="16"
                viewBox="0 0 18 18"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="black"
                >
                  <path
                    d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
        <div
          class="ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 undefined undefined"
        >
          Foo
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Modal CSS API default 1`] = `
<div>
  <div
    class="flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill items-center"
    data-testid="modal__overlay"
    tabindex="-1"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      class="undefined"
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="vtex-modal__title"
        aria-modal="true"
        class="flex flex-column relative bg-white shadow-5 center mv9 vw-50-ns vw-70"
        data-testid="modal__modal"
        role="dialog"
      >
        <div
          class="mb5"
        >
          <div
            class="min-h-small min-h-large-ns pl6 pl8-ns"
          >
            <div
              class="fr pointer pr6-ns pt6-ns pr5 pt5"
              role="button"
              tabindex="0"
            >
              <svg
                class="vtex__icon-close  "
                fill="none"
                height="16"
                viewBox="0 0 18 18"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="black"
                >
                  <path
                    d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
        <div
          class="ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 undefined undefined"
        >
          Foo
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Modal CSS API responsiveFullScreen true 1`] = `
<div>
  <div
    class="flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill items-center"
    data-testid="modal__overlay"
    tabindex="-1"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      class="undefined"
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="vtex-modal__title"
        aria-modal="true"
        class="flex flex-column relative bg-white shadow-5 center mv9 vw-50-ns h-100 h-auto-ns vw-100"
        data-testid="modal__modal"
        role="dialog"
      >
        <div
          class="mb5"
        >
          <div
            class="min-h-small min-h-large-ns pl6 pl8-ns"
          >
            <div
              class="fr pointer pr6 pt6"
              role="button"
              tabindex="0"
            >
              <svg
                class="vtex__icon-close  "
                fill="none"
                height="16"
                viewBox="0 0 18 18"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="black"
                >
                  <path
                    d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
        <div
          class="ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 undefined undefined"
        >
          Foo
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Modal CSS API showBottomBarBorder false 1`] = `
<div>
  <div
    class="flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill items-center"
    data-testid="modal__overlay"
    tabindex="-1"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      class="undefined"
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="vtex-modal__title"
        aria-modal="true"
        class="flex flex-column relative bg-white shadow-5 center mv9 vw-50-ns vw-70"
        data-testid="modal__modal"
        role="dialog"
      >
        <div
          class="mb5"
        >
          <div
            class="min-h-small min-h-large-ns pl6 pl8-ns"
          >
            <div
              class="fr pointer pr6-ns pt6-ns pr5 pt5"
              role="button"
              tabindex="0"
            >
              <svg
                class="vtex__icon-close  "
                fill="none"
                height="16"
                viewBox="0 0 18 18"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="black"
                >
                  <path
                    d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
        <div
          class="ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 undefined undefined"
        >
          Foo
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Modal CSS API showTopBar false 1`] = `
<div>
  <div
    class="flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill items-center"
    data-testid="modal__overlay"
    tabindex="-1"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      class="undefined"
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="vtex-modal__title"
        aria-modal="true"
        class="flex flex-column relative bg-white shadow-5 center mv9 vw-50-ns vw-70"
        data-testid="modal__modal"
        role="dialog"
      >
        <div
          class="mb5"
        >
          <div
            class="min-h-small min-h-large-ns pl6 pl8-ns"
          >
            <div
              class="fr pointer pr6-ns pt6-ns pr5 pt5"
              role="button"
              tabindex="0"
            >
              <svg
                class="vtex__icon-close  "
                fill="none"
                height="16"
                viewBox="0 0 18 18"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="black"
                >
                  <path
                    d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
        <div
          class="ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 undefined undefined"
        >
          Foo
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Modal bottomBar should render bottomBar 1`] = `
<div>
  <div
    class="flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill items-center"
    data-testid="modal__overlay"
    tabindex="-1"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      class="undefined"
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="vtex-modal__title"
        aria-modal="true"
        class="flex flex-column relative bg-white shadow-5 center mv9 vw-50-ns vw-70"
        data-testid="modal__modal"
        role="dialog"
      >
        <div
          class="mb5"
        >
          <div
            class="min-h-small min-h-large-ns pl6 pl8-ns"
          >
            <div
              class="fr pointer pr6-ns pt6-ns pr5 pt5"
              role="button"
              tabindex="0"
            >
              <svg
                class="vtex__icon-close  "
                fill="none"
                height="16"
                viewBox="0 0 18 18"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="black"
                >
                  <path
                    d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
        <div
          class="ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 undefined undefined"
        >
          Foo
        </div>
        <div
          class="flex justify-content flex-row-reverse min-h-regular-ns min-h-small pv6-ns ph8-ns pv7 ph6 mt3 bt b--muted-4"
        >
          <div>
            <div>
              Bar
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Modal container should render modal inside it 1`] = `
<div>
  <div
    class="flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill items-center"
    data-testid="modal__overlay"
    tabindex="-1"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      class="undefined"
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="vtex-modal__title"
        aria-modal="true"
        class="flex flex-column relative bg-white shadow-5 center mv9 vw-50-ns vw-70"
        data-testid="modal__modal"
        role="dialog"
      >
        <div
          class="mb5"
        >
          <div
            class="min-h-small min-h-large-ns pl6 pl8-ns"
          >
            <div
              class="fr pointer pr6-ns pt6-ns pr5 pt5"
              role="button"
              tabindex="0"
            >
              <svg
                class="vtex__icon-close  "
                fill="none"
                height="16"
                viewBox="0 0 18 18"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="black"
                >
                  <path
                    d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
        <div
          class="ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 undefined undefined"
        >
          Foo
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Modal isOpen Should show modal when isOpen is true and hide when is false 1`] = `
<div>
  <div
    class="flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill items-center"
    data-testid="modal__overlay"
    tabindex="-1"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      class="undefined"
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="vtex-modal__title"
        aria-modal="true"
        class="flex flex-column relative bg-white shadow-5 center mv9 vw-50-ns vw-70"
        data-testid="modal__modal"
        role="dialog"
      >
        <div
          class="mb5"
        >
          <div
            class="min-h-small min-h-large-ns pl6 pl8-ns"
          >
            <div
              class="fr pointer pr6-ns pt6-ns pr5 pt5"
              role="button"
              tabindex="0"
            >
              <svg
                class="vtex__icon-close  "
                fill="none"
                height="16"
                viewBox="0 0 18 18"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="black"
                >
                  <path
                    d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
        <div
          class="ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 undefined undefined"
        >
          Foo
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Modal isOpen Should show modal when isOpen is true and hide when is false 2`] = `
<div>
  <div
    class="flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill items-center"
    data-testid="modal__overlay"
    tabindex="-1"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      class="undefined"
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="vtex-modal__title"
        aria-modal="true"
        class="flex flex-column relative bg-white shadow-5 center mv9 vw-50-ns vw-70"
        data-testid="modal__modal"
        role="dialog"
      >
        <div
          class="mb5"
        >
          <div
            class="min-h-small min-h-large-ns pl6 pl8-ns"
          >
            <div
              class="fr pointer pr6-ns pt6-ns pr5 pt5"
              role="button"
              tabindex="0"
            >
              <svg
                class="vtex__icon-close  "
                fill="none"
                height="16"
                viewBox="0 0 18 18"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="black"
                >
                  <path
                    d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  />
                </g>
              </svg>
            </div>
          </div>
        </div>
        <div
          class="ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 undefined undefined"
        >
          Foo
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Modal title should render title prop 1`] = `
<DocumentFragment>
  <div
    class="flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill items-center"
    data-testid="modal__overlay"
    tabindex="-1"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      class="undefined"
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="vtex-modal__title"
        aria-modal="true"
        class="flex flex-column relative bg-white shadow-5 center mv9 vw-50-ns vw-70"
        data-testid="modal__modal"
        role="dialog"
      >
        <div
          class="mb5"
        >
          <div
            class="min-h-small min-h-large-ns pl6 pl8-ns"
          >
            <div
              class="fr pointer pr6-ns pt6-ns pr5 pt5"
              role="button"
              tabindex="0"
            >
              <svg
                class="vtex__icon-close  "
                fill="none"
                height="16"
                viewBox="0 0 18 18"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="black"
                >
                  <path
                    d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  />
                </g>
              </svg>
            </div>
          </div>
          <span
            class="t-heading-4 ml8"
          >
            <div
              class="undefined"
              id="vtex-modal__title"
            >
              modal title
            </div>
          </span>
        </div>
        <div
          class="ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 undefined undefined"
        >
          Foo
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</DocumentFragment>
`;
